src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Generics.cs

```diff
@@ -59,40 +63,62 @@ internal sealed partial class ExpressionEval {
         }
 
         /// <summary>
-        /// Given generic type and list of indices in the expression like
-        /// Generic[T1, T2, ...] or List[str] creates generic class base
-        /// (if the former) on specific type (if the latter).
+        /// Returns whether the arguments to Generic are valid
         /// </summary>
-        private IMember CreateSpecificTypeFromIndex(IGenericType gt, IReadOnlyList<IMember> indices, Expression expr) {
-            // See which ones are generic parameters as defined by TypeVar() 
-            // and which are specific types. Normally there should not be a mix.
-            var genericTypeArgs = indices.OfType<IGenericTypeDefinition>().ToArray();
-            var specificTypes = indices.Where(i => !(i is IGenericTypeDefinition)).OfType<IPythonType>().ToArray();
-
-            if (genericTypeArgs.Length > 0 && genericTypeArgs.Length != indices.Count) {
-                // TODO: report that some type arguments are not declared with TypeVar.
+        private bool GenericClassParameterValid(IGenericTypeDefinition[] genericTypeArgs, IReadOnlyList<IMember> indices, Expression expr) {
+            if (genericTypeArgs.Length == 0) {
+                return false;
             }
-            if (specificTypes.Length > 0 && specificTypes.Length != indices.Count) {
```

Comment:
Really wasn't sure what error this TODO was talking about, could someone explain?

Reply:
Is this question still active?

Reply:
I think it was just an oversight, I don't think it actually matters.